com.microsoft.mrtk.input/Interactors/InteractorVisuals/MRTKRayReticleVisual.cs

```diff
@@ -71,50 +70,53 @@ private void UpdateReticle()
-                    rayInteractor.enableUIInteraction && rayInteractor.TryGetCurrentUIRaycastResult(out _);
-
-                if (showReticle)
+                if (Reticle != null)
                 {
-                    if (rayInteractor.interactablesSelected.Count > 0)
-                    {
-                        reticlePosition = hitTargetTransform.TransformPoint(targetLocalHitPoint);
-                        reticleNormal = hitTargetTransform.TransformDirection(targetLocalHitNormal);
-                        Reticle.SetActive(true);
-                    }
-                    else
-                    {
-                        bool rayHasHit = rayInteractor.TryGetHitInfo(out reticlePosition, out reticleNormal, out int _, out bool _);
-                        Reticle.SetActive(rayHasHit);
-                    }
-
-                    // Ensure that our visuals position and normal are set correctly.
-                    // The reticle should be a direct child of this gameobject, so it's position and rotation should match this gameobject's
-                    transform.position = reticlePosition;
-                    transform.forward = reticleNormal;
+                    bool showReticle = VisibilitySettings == ReticleVisibilitySettings.AllValidSurfaces || rayInteractor.hasHover || rayInteractor.hasSelection ||
+                        rayInteractor.enableUIInteraction && rayInteractor.TryGetCurrentUIRaycastResult(out _);
 
-                    // If the reticle is an IVariableSelectReticle, have the reticle update based on selectedness
-                    if (variableReticle != null)
+                    if (showReticle)
                     {
-                        if (rayInteractor is IVariableSelectInteractor variableSelectInteractor)
+                        if (rayInteractor.interactablesSelected.Count > 0)
                         {
-                            variableReticle.UpdateVisuals(variableSelectInteractor.SelectProgress);
+                            reticlePosition = hitTargetTransform.TransformPoint(targetLocalHitPoint);
+                            reticleNormal = hitTargetTransform.TransformDirection(targetLocalHitNormal);
+                            Reticle.SetActive(true);
                         }
                         else
                         {
-                            variableReticle.UpdateVisuals(rayInteractor.isSelectActive ? 1 : 0);
+                            bool rayHasHit = rayInteractor.TryGetHitInfo(out reticlePosition, out reticleNormal, out int _, out bool _);
+                            Reticle.SetActive(rayHasHit);
+                        }
+
+                        // Ensure that our visuals position and normal are set correctly.
+                        // The reticle should be a direct child of this gameobject, so it's position and rotation should match this gameobject's
+                        reticleRoot.transform.position = reticlePosition;
```

Comment:
Should we null check `reticleRoot` and/or auto-set it to some reasonable value if it's null in `OnEnable()` or something?

Reply:
Yeah, I'll do this in a PR which needs to address other reticle issues raised